deepspeed/runtime/engine.py

```diff
@@ -1014,8 +1014,8 @@ def _do_sanity_check(self):
                     self.optimizer_name()), "{} is not a supported DeepSpeed Optimizer".format(self.optimizer_name())
 
         if (self.optimizer_name() == LAMB_OPTIMIZER or self.optimizer_name() == ONEBIT_LAMB_OPTIMIZER):
-            assert (self.dynamic_loss_scale()), "DeepSpeed {} optimizer requires dynamic loss scaling".format(
-                self.optimizer_name())
+            assert (self.dynamic_loss_scale() and not self.bfloat16_enabled()
```

Comment:
Will try switching to check the optimizer wrapper as BFLOAT16 rather than just checking if bfloat is enabled